classes/SpecificPrice.php

```diff
@@ -89,6 +89,8 @@ class SpecificPriceCore extends ObjectModel
 
 
     protected static $_specificPriceCache = array();
+    protected static $_couldHaveSpecificPriceCache = array();
+    protected static $_hasGlobalProductRules = null;
```

Comment:
Please add a PhpDoc header on your new properties.

It is important that each property describes:
- A short summary of what it holds / its purpose
- Type

Documenting properties is important because
- It enables autocompletion by IDEs
- It eases code understanding

Example:
```php
   /**
     * Query string parameters ($_GET).
     *
     * @var \Symfony\Component\HttpFoundation\ParameterBag
     */
    public $query;
```

You can find out more about PhpDoc by reading the [proposed PSR-5 standard](https://github.com/php-fig/fig-standards/blob/master/proposed/phpdoc.md).